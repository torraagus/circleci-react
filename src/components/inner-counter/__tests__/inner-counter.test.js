import React from 'react';
import ReactDOM from 'react-dom';
import InnerCounter from './../../inner-counter/inner-counter.component';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from 'react-test-renderer';

afterEach(cleanup);

test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<InnerCounter></InnerCounter>, div);
})

it('matches snapshot', () => {
    const tree = renderer.create(<InnerCounter counter="1"></InnerCounter>).toJSON();
    expect(tree).toMatchSnapshot();
})
