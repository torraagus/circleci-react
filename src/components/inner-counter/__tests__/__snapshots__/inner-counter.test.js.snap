// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="grid-counter-inner container-fluid"
>
  <div
    className="row center-xs"
  >
    <div
      className="col-xs-12 p-0"
    >
      <h4
        className="counterName-inner p-2 mb-0"
      >
        Inner Counter
      </h4>
    </div>
  </div>
  <div
    className="row between-xs"
  >
    <div
      className=""
    >
      <button
        className="increaseBtnSmall"
        onClick={[Function]}
      >
        +
      </button>
    </div>
    <div
      className="counter-inner"
    >
      1
    </div>
    <div
      className=""
    >
      <button
        className="decreaseBtnSmall"
        onClick={[Function]}
      >
        -
      </button>
    </div>
  </div>
</div>
`;
